src/main/java/net/skinsrestorer/bukkit/SkinsRestorer.java

```diff
@@ -446,12 +447,12 @@ public SkinsRestorerBukkitAPI(MojangAPI mojangAPI, SkinStorage skinStorage) {
 
         @Override
         public void applySkin(PlayerWrapper player, IProperty props) {
-            getSkinApplierBukkit().applySkin(player.get(Player.class), props);
+            skinApplierBukkit.applySkin(player.get(Player.class), props);
         }
 
         @Override
-        public void applySkin(PlayerWrapper player) {
-            getSkinApplierBukkit().applySkin(player.get(Player.class), getSkinData(getSkinName(player.get(Player.class).getName())));
+        public void applySkin(PlayerWrapper player) throws SkinRequestException {
+            applySkin(player, skinStorage.getOrCreateSkinForPlayer(player.get(Player.class).getName(), false));
```

Comment:
applySkin should not use getOrCreateSkinForPlayer
As this will create and or update a skin.